plugins/notifications-backend/src/service/router.ts

```diff
@@ -1,61 +1,160 @@
 import { errorHandler } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
 
 import express from 'express';
 import Router from 'express-promise-router';
-import { Knex } from 'knex';
+import knex from 'knex';
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
 
 export interface RouterOptions {
   logger: Logger;
-  dbClient?: Knex<any, any[]>;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
 }
 
+export type CreateRequest = {
+  subject: string;
+  body?: string;
+  targetUsers?: string[];
+  targetGroups?: string[];
+}
+
+
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbClient } = options;
+  const { logger, dbConfig, catalogClient } = options;
 
   const router = Router();
   router.use(express.json());
 
-  if (!dbClient) {
-    logger.error('Missing dbClient');
+  if (!dbConfig) {
+    logger.error('Missing dbConfig');
     return router;
   }
 
-  router.get('/health', (_, response) => {
-    logger.info('PONG!');
-    response.json({ status: 'ok' });
+  const dbClient = knex({
+    client: dbConfig.getString('client'),
+    connection: {
+      host: dbConfig.getString('connection.host'),
+      port: dbConfig.getNumber('connection.port'),
+      user: dbConfig.getString('connection.user'),
+      password: dbConfig.getString('connection.password'),
+      database: 'backstage_plugin_notifications'
+    }
+  });
+
+  if (! await dbClient.schema.hasTable('messages')) {
+    await dbClient.schema.createTable('messages', function (table) {
+      table.uuid("id", {primaryKey: true}).defaultTo(dbClient.fn.uuid());
```

Comment:
The UUID is a `string`. I saw it re-typed to `number`

Reply:
retyped where?

Reply:
I think it's outdated since the time of writing the comment